pywbem/_subscription_manager.py

```diff
@@ -343,27 +343,35 @@ def add_server(self, server):
         self._owned_subscriptions[server_id] = []
         self._owned_filters[server_id] = []
         self._owned_destinations[server_id] = []
+        # self._systemnames[server_id] = None
+
+        # TODO expand this to account for WBEM Server profile.
 
         # Recover any owned destination, filter, and subscription instances
         # that exist on this server
-
-        this_host = getfqdn()
-
         dest_name_pattern = re.compile(
-            _format(r'^pywbemdestination:owned:{0}:{1}:[^:]*$',
-                    this_host, self._subscription_manager_id))
+            _format(r'^pywbemdestination:owned:{0}:[^:]*$',
+                    self._subscription_manager_id))
         dest_insts = server.conn.EnumerateInstances(
             DESTINATION_CLASSNAME, namespace=server.interop_ns)
-        for inst in dest_insts:
-            if re.match(dest_name_pattern, inst.path.keybindings['Name']) \
-                    and inst.path.keybindings['SystemName'] == this_host:
-                self._owned_destinations[server_id].append(inst)
+        if dest_insts:
+            self._systemnames[server_id] = dest_insts[0]['SystemName']
+            this_host = self._systemnames[server_id]
```

Comment:
I suggest renaming this variable to something like server_host or system_name, since this_host was meant to be the host the client runs on